### ISSUE TITLE
Does not work without user signed into Mac GUI

### ISSUE THREAD

**Issue by mnsanghvi** [mas-cli/mas#266]
## Your Environment

- `mas version`: 1.6.3
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 10.15.4 (19E287)

### mas Install Method

- [x] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
   - Fork/branch: ? (e.g. mas-cli/master)
   - Xcode version: 10.1

## Describe the Bug

> A clear and concise description of what the bug is.
- I login via SSH to my iMac
```
mehuls-iMac:~ mehul$ sw_vers 
ProductName:    Mac OS X
ProductVersion: 10.15.4
mehuls-iMac:~ mehul$ mas version
1.6.3
mehuls-iMac:~ mehul$ echo "No MAS GUI Open, user logged out on iMac" ; mas list ; mas outdated ; mas upgrade
No MAS GUI Open, user logged out on iMac
No installed apps found
Everything is up-to-date
mehuls-iMac:~ mehul$ echo "Test #2: No MAS GUI Open, user logged into console on iMac" ; mas list ; mas outdated ; mas upgrade 
Test #2: No MAS GUI Open, user logged into console on iMac
405399194 Kindle (1.28.0)
497799835 Xcode (10.1)
409183694 Keynote (10.0)
408981434 iMovie (10.1.14)
408981381 iPhoto (9.6.1)
715768417 Microsoft Remote Desktop (8.0.30030)
1432731683 Adblock Plus (2.1.1)
431469123 MahJongFree (2.0)
928590988 Getty Images Stream (1.0.0)
409201541 Pages (10.0)
408980954 GarageBand (6.0.5)
409203825 Numbers (10.0)
Everything is up-to-date
mehuls-iMac:~ mehul$ 
```

MAS GUI shows that there is Xcode 11.4 available to replace Xcode 10.1.

The same results as Test #2 occur when I run the commands via Terminal.app on the iMac.

## To Reproduce

Steps provided above.

## Expected Behavior

> A clear and concise description of what you expected to happen.

Since this is a CLI tool meant for automation and scripting, I would expect to be able to run the tool via SSH. 
I would expect `mas list` to provide a list of installed applications without needing MAS GUI to be open.
I would expect `mas outdated` to list the applications that need to be upgraded, in this case Xcode.
I would expect `mas upgrade` to upgrade the applications with a pending update

## Actual Behavior

Nothing happened.

> Did it work in a previous version?
First time using it

This might be similar to #248, #71, and #244

**Comment by phatblat** [comment]
The `list` command may have been affected by a bug fixed recently in 1.6.4. Can you retry on this update and let me know if the situation has changed?

As for `outdated` and `upgrade`, these have been broken for a while and I don't have a handle on how to fix them yet.

Using mas over SSH often has issues if there is no user logged into the macOS GUI. I suspect that this is due to the App Store never being designed to operate in the background as a service; normally users log in first before they launch the app. However, the underlying pieces of the App Store have been changing (this is what broke the `outdated` and `upgrade` commands) and they are now implemented as XPC services. While they pose a challenge to integrate with, the good news about this service architecture is that it may work much better without a user logged in - once I can figure out how to use them. 😉

**Comment by chris-araman** [comment]
Closing after no response.
